### DISCUSSION TITLE
DataGrid with groups?

### DISCUSSION THREAD

**Discussion by sn4k3** [AvaloniaUI/Avalonia#6612] [Q&A] [answered]
Is it possible to have groups on datagrid by a column?
Can't find any sample. But i find this image on internet which is exactly want i need:

![image](https://user-images.githubusercontent.com/113281/133910576-448d85d6-cff1-4034-99ed-dea6970b89a1.png)

**Comment by maxkatz6** [comment, marked as answer]
There is an example in ControlsCatalog data grid page

**Reply by sn4k3** [reply]
Works, thanks!

**Comment by sn4k3** [comment]
I might found a bug, but needs confirming...

I have the following code:

```C#
        public IEnumerable IssuesGridItems 
        {
            get
            {
                if (!IsFileLoaded) return null;
                if (Settings.Issues.DataGridGroupByType || Settings.Issues.DataGridGroupByLayerIndex)
                {
                    var groupView = new DataGridCollectionView(SlicerFile.IssueManager);
                    if (Settings.Issues.DataGridGroupByType) groupView.GroupDescriptions.Add(new DataGridPathGroupDescription("Type"));
                    if (Settings.Issues.DataGridGroupByLayerIndex) groupView.GroupDescriptions.Add(new DataGridPathGroupDescription("StartLayerIndex"));

                    return groupView;
                }

                return SlicerFile.IssueManager;
            }
        }
```

It works as expected, but when it return groups it auto select the first index on the list. While if it ungrouped, it populate the list without selected any item (The correct way)

To reproduce is easy as call `RaisePropertyChanged(nameof(IssuesGridItems));`. Additionally this don't happens if we Replace the item collection on the source variable (`Items.Clear(); Items.AddRange(...)`). I make further testing and instead of always create a new group, i just return the same instance already grouped, and the problem still happens. So no matter what i try, always the DataGrid get notified for `.Items` it selects the first index on the list (0) instead of (-1)
I also tried to force `SelectedIndex = -1;` before return the Items, but still no effect. 
This is a problem in my app because on item change it change layer and viewport of image when we need to refresh the grid.

![2021-09-21_01-50-48](https://user-images.githubusercontent.com/113281/134096236-1d06ae58-d137-4c58-99c0-254ba1bbba38.gif)

**Reply by FoggyFinder** [reply]
can you create a separate issue for this? Best if you provide MCVE there

**Reply by sn4k3** [reply]
> can you create a separate issue for this? Best if you provide MCVE there

Will do
